openapi/openapi.go

```diff
@@ -55,16 +55,18 @@ func (o *OpenAPI) SetChild(i string, child tree.NodeTraverser) {
 	errors.CastFail("(o *OpenAPI) setChild", "NodeTraverser", "*schema.Schema")
 }
 
-// LoadFromYaml expects an absolute path. It will unmarshal the yaml file and resolve the references contained within.
+// LoadFromYaml expects a path to an openapi file in yaml format. It will unmarshal the yaml file and resolve the
+// references contained within.
 func LoadFromYaml(openApiFilepath string) (*OpenAPI, error) {
 	// Read yaml file
+	// TODO: should this function have to perform file IO?
```

Comment:
If you take file IO out of the equation, then this function becomes nothing but an indirection layer to api.LoadFromBytes. We could just call api.LoadFromBytes and eliminate this function if we remove file IO.

I'm not opposed to that.

Another option would be to keep this function as a convenience that deals with File IO and to call api.LoadFromBytes where we don't need file IO. If we choose to do so, I think it would be good to rename this function to refer to the fact that it loads from a file specifically.

Reply:
I'll make sure to add the name change into the next PR. `LoadFromFile()` feels most appropriate. 
It's intended to collect the logic provided by the three private functions used and expose it as a public function, so the indirection layer you make reference to is exactly as intended